core/src/main/scripts/importer/importOncokbDiscreteCNA.py

```diff
@@ -0,0 +1,288 @@
+            continue  # skip comment and header line
+        line_elements = line.rstrip('\n').split('\t')
+        for sample_id in sample_ids:
+            feature = {}
+            feature['sample_id'] = sample_id
+            feature['alteration'] = int(line_elements[sample_indexes[sample_id]])
+            # cna value 0 (no CNA) is skipped
+            if feature['alteration'] == 0:
+                continue
+            feature['copyNameAlterationType'] = list(cna_alteration_types.keys())[
+                list(cna_alteration_types.values()).index(feature['alteration'])]
+            for column_name, index in header_indexes.items():
+                value = line_elements[index]
+                if value != '':
+                    feature[column_name] = value
+                elif column_name != 'Entrez_Gene_Id':
+                    print(Color.RED + "Empty value encounterd in column '" + column_name + "' in row " + str(
+                        row_counter) + ". OncoKb annotations cannot be imported. Please fix and rerun." + Color.END,
+                          file=sys.stderr)
+                    print("!" * 71, file=sys.stderr)
+                    sys.exit(1)
+
+            # resolve gene symbols to Entrez Ids if needed
+            if 'Entrez_Gene_Id' in feature and feature['Entrez_Gene_Id'] is not None and feature[
+                'Entrez_Gene_Id'] != '':
+                entrez_gene_ids = [feature['Entrez_Gene_Id']]
+            else:
+                entrez_gene_ids = portal_instance.hugo_entrez_map[feature['Hugo_Symbol']]
+
+            if len(entrez_gene_ids) > 1:
+                logger.error("Multiple Entrez gene ids were found for a gene." \
+                             "OncoKb annotations will not be imported for this gene." \
+                             "Please fix and rerun.",
+                             extra={'symbol': feature['Hugo_Symbol']})
+                feature['Entrez_Gene_Id'] = None
+            elif len(entrez_gene_ids) == 0:
+                logger.error("Could not find the Entrez gene id for a gene." \
+                             "OncoKb annotations will not be imported for this gene." \
+                             "Please fix and rerun.",
+                             extra={'symbol': feature['Hugo_Symbol']})
+                feature['Entrez_Gene_Id'] = None
+            else:
+                feature['Entrez_Gene_Id'] = str(entrez_gene_ids[0])
+                feature['id'] = "_".join([feature['Entrez_Gene_Id'], feature['copyNameAlterationType']])
+
+            features.append(feature)
+    cna_file.close()
+    print(" DONE")
+    return features
+
+
+def fetch_oncokb_annotations(features):
+    """Submit CNA events to OncoKB.org and return OncoKB annotations."""
+    id_to_annotation = {}
+    payload_list = create_request_payload(features)
+    annotations = libImportOncokb.fetch_oncokb_annotations(payload_list, "https://public.api.oncokb.org/api/v1/annotate/copyNumberAlterations")
```

Comment:
Can we move the `public.api.oncokb.org/api/v1` to a global constant so it can be shared?

Reply:
Done